samples/react-tabacordion/src/webparts/tabAccordion/TabAccordionWebPart.ts

```diff
@@ -0,0 +1,205 @@
+import { ICAccordionProps } from './components/ICAccordionProps';
+import { SPComponentLoader } from '@microsoft/sp-loader';
+import 'tinymce';
+export interface ITabAccordionWebPartProps {
+  tabs: any[]; 
+  type: string;
+  title: string;
+  accordion:boolean;
+  tabContent: string;
+}
+
+export default class TabAccordionWebPart extends BaseClientSideWebPart<ITabAccordionWebPartProps> {
+  private propertyFieldCollectionData;
+  private customCollectionFieldType;
+  private guid: string;
+  private isMobile: boolean;
+
+  // ...
+  protected get propertiesMetadata(): IWebPartPropertiesMetadata {
+    return {
+      'title': { isSearchablePlainText: true },
+      'tabContent': { isHtmlString: true }
+    };
+  }
+  // ...
+
+  /**
+   * @function
+   * Web part contructor.
+   */
+  public constructor(context?: WebPartContext) {
+    super();
+
+    //Initialize unique GUID
+    this.guid = this.getGuid();
+
+    this.isMobile = this.detectmob();
+
+    //Hack: to invoke correctly the onPropertyChange function outside this class
+    //we need to bind this object on it first
+    this.onPropertyPaneFieldChanged = this.onPropertyPaneFieldChanged.bind(this);
+  }
+  
+
+  public render(): void {
+    console.log('Web Part Render Called');
+    this.properties.tabContent = "";
+    this.properties.tabs.map((tab: any, tabindex: number) => {
+      this.properties.tabContent += tab.Title + "," + tab.Content + "|";
+    });
+    
+    const elementTab: React.ReactElement<ICTabProps > = React.createElement(
+      Tab,
+      {        
+        tabs: this.properties.tabs, 
+        displayMode: this.displayMode,
+        guid: this.guid,
+        title:this.properties.title
+      }
+    );
+    const elementAccordion: React.ReactElement<ICAccordionProps > = React.createElement(
+      Accordion,
+      {        
+        tabs: this.properties.tabs, 
+        displayMode: this.displayMode,
+        guid: this.guid,
+        title: this.properties.title,
+        accordion:this.properties.accordion
+      }
+    );
+    if(this.isMobile)
+    {
+      ReactDom.render(elementAccordion, this.domElement);
+    }
+    else 
+    {
+      if(this.properties.type == "Accordion")
+      {
+        ReactDom.render(elementAccordion, this.domElement);
+      }
+      else
+      {
+        ReactDom.render(elementTab, this.domElement);
+      }
+    }
+    
+  }
+
+  protected get dataVersion(): Version {
+    return Version.parse('1.0');
+  }
+
+  /**
+   * @function
+   * Generates a GUID
+   */
+  private getGuid(): string {
+    return this.s4() + this.s4() + '-' + this.s4() + '-' + this.s4() + '-' +
+      this.s4() + '-' + this.s4() + this.s4() + this.s4();
+  }
+
+  private detectmob(): boolean {
+    console.log('inside detectmob');
+    if(window.innerWidth <= 480) {
+      return true;
+    } else {
+      return false;
+    }
+ }
+
+  /**
+   * @function
+   * Generates a GUID part
+   */
+  private s4(): string {
```

Comment:
This won't be needed if you use the built in `Guid` functions

Reply:
Use the built in Guid